internal/kafka/internal/services/data_plane_kafka.go

```diff
@@ -150,30 +151,57 @@ func (d *dataPlaneKafkaService) processRealKafkaDeployment(ks *dbapi.DataPlaneKa
+	//  - 'suspended' state must only transition to 'resuming' or 'deprovision'.
+	//  - 'deprovision' state is set by the user. This must only transition to 'deleting' or 'failed'.
+	//     - FSO will only send the status 'deleted' if kafka was already in a 'deprovisioning' state.
+	//  - 'failed' (or 'error') state may occur at any time. However, KFM must never transition a Kafka instance to 'failed' from 'suspending' and 'suspended' states.
 	var e *serviceError.ServiceError
 	switch s := d.getManagedKafkaStatus(ks); s {
 	case statusReady:
-		// Store the routes (and create them) when Kafka is ready. By the time it is ready, the routes should definitely be there.
-		e = d.persistKafkaRoutes(kafka, ks, cluster)
-		if e == nil {
-			kafka.AdminApiServerURL = ks.AdminServerURI
-			e = d.setKafkaClusterReady(kafka)
+		if kafka.Status != constants2.KafkaRequestStatusSuspending.String() {
+			// Store the routes (and create them) when Kafka is ready. By the time it is ready, the routes should definitely be there.
+			e = d.persistKafkaRoutes(kafka, ks, cluster)
+			if e == nil {
+				kafka.AdminApiServerURL = ks.AdminServerURI
+				e = d.setKafkaClusterReady(kafka)
+			}
 		}
 	case statusInstalling:
-		// Store the routes (and create them) if they are available at this stage to lessen the length of time taken to provision the Kafka.
-		// The routes list will either be empty or complete.
-		e = d.persistKafkaRoutes(kafka, ks, cluster)
+		if kafka.Status != constants2.KafkaRequestStatusSuspending.String() {
+			// Store the routes (and create them) if they are available at this stage to lessen the length of time taken to provision the Kafka.
+			// The routes list will either be empty or complete.
+			e = d.persistKafkaRoutes(kafka, ks, cluster)
+		}
 	case statusError:
-		// when getStatus returns statusError we know that the ready
-		// condition will be there so there's no need to check for it
-		readyCondition, _ := ks.GetReadyCondition()
-		e = d.setKafkaClusterFailed(kafka, readyCondition.Message)
+		if kafka.Status != constants2.KafkaRequestStatusSuspending.String() || kafka.Status != constants2.KafkaRequestStatusSuspended.String() {
+			// when getStatus returns statusError we know that the ready
+			// condition will be there so there's no need to check for it
+			readyCondition, _ := ks.GetReadyCondition()
+			e = d.setKafkaClusterFailed(kafka, readyCondition.Message)
+		}
 	case statusDeleted:
 		e = d.setKafkaClusterDeleting(kafka)
 	case statusRejected:
-		e = d.reassignKafkaCluster(kafka)
+		if kafka.Status != constants2.KafkaRequestStatusSuspending.String() {
+			e = d.reassignKafkaCluster(kafka)
+		}
 	case statusRejectedClusterFull:
-		e = d.unassignKafkaFromDataplaneCluster(kafka)
+		if kafka.Status != constants2.KafkaRequestStatusSuspending.String() {
+			e = d.unassignKafkaFromDataplaneCluster(kafka)
+		}
```

Comment:
Is it possible for fleetshard to report these two statuses when a Kafka is in a `suspended` state?

Reply:
Judging from the capacity reporting conversation, when the Kafka is in `provisioning` state, it'll first report rejected due to fullness before the cluster is full. @MikeEdgar is this something that going to be honored? 

> Should we allow Kafkas to be re-assigned to another cluster when it's in a suspending or suspended state?

No, we shouldn't support for this as we risk loosing data if the kafka state was ready before it was suspended. 
Based on what I am saying above, we shouldn't really be concerned about re-assigning the suspended/suspending kafka onto another cluster. We could add a guard against these states.

Reply:
The rejected status will take priority over the suspended status. I think that from KFM's perspective, it may appear that the cluster transitions from installing -> suspending -> rejected. In that case FSO would only notify of the rejection and those other status values would originate in the control plane.

Reply:
if we don't allow for re-assignment, how would this affect the upgrade? In Data Plane, FSO will resume suspended Kafkas in order to upgrade it (while continuing to report status reason as 'Suspended'). If there's no capacity on the cluster, would this mean the upgrade will always fail?

Reply:
> I wonder to avoid this scenario, if it is the case, would it be best to not be able to suspend a Kafka instance unless it was in a ready state previously?

I think this makes sense. What do you think? @MikeEdgar

Reply:
cc @pawelpaszki

Reply:
This was addressed #1344. Thanks guys!